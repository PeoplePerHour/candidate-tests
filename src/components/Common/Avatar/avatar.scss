@import '~styles/colors';

.Avatar {
  width: 120px;
  height: 120px;
  background-color: transparent;
  border-radius: 50%;
  max-width: 120px;
  max-height: 120px;
}

.Avatar__image-container {
  position: relative;
  width: 100%;
  height: 100%;

  > img {
    border: 1px solid rgba($gray,0.5);
    border-radius: 50%;
    bottom: 0;
    height: auto;
    left: 0;
    position: absolute;
    right: 0;
    top: 0;
    width: 100%;
  }
}

.Avatar--square {
  width: 200px;
  height: 200px;
  max-width: 200px;
  max-height: 200px;
  border-radius: 0;

  img {
    width: 200px;
    height: 200px;
    border-radius: 0;
  }
}
