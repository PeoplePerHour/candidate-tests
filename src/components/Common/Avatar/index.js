import React from 'react';

import cn from 'classnames';
import './avatar.scss';

const Avatar = ({ src, alt, square = false }) => (
  <div className={cn('Avatar', { 'Avatar--square': square })}>
    <div className="Avatar__image-container">
      <img src={src} alt={alt ? alt : 'avatar'} />
    </div>
  </div>
);

export default Avatar;
