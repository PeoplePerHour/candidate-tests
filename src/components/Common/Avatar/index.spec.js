import React from 'react';
import renderer from 'react-test-renderer';
import Avatar from './';

it('renders correctly', () => {
  const tree = renderer
    .create(<Avatar src="https://placekitten.com/200/300" alt="kitten" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
