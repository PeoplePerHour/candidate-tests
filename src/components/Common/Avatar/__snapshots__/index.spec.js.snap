// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Avatar"
>
  <div
    className="Avatar__image-container"
  >
    <img
      alt="kitten"
      src="https://placekitten.com/200/300"
    />
  </div>
</div>
`;
